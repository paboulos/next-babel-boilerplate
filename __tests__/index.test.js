/* eslint-env jest */

import { shallow } from 'enzyme'
import React from 'react'
import renderer from 'react-test-renderer'

import App from '../pages/index.js'

describe('With Enzyme', () => {
  it('App shows "Hello World"', () => {
    const app = shallow(<App />)

    expect(app.find('a').text()).toEqual('View Settings')
  })
})

describe('With Snapshot Testing', () => {
  it('App shows "Hello World"', () => {
    const component = renderer.create(<App />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
