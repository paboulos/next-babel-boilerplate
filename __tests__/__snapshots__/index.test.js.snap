// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing App shows "Hello World" 1`] = `
<div
  className="block"
>
  <h1
    className="text-center"
  >
    <span
      className="block text-gray leading-tight"
    >
      Welcome to the
    </span>
    <span
      className="block bold leading-none"
    >
      Web App
    </span>
  </h1>
  <div
    className="text-center"
  >
    <a
      className="inline-block example hover:black"
      href="/submenu-one/tab-one"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      View Settings
    </a>
  </div>
</div>
`;
